deepspeed/runtime/engine.py

```diff
@@ -458,7 +458,14 @@ def _set_distributed_vars(self):
 
     # Configure based on command line arguments
     def _configure_with_arguments(self, args, mpu):
-        self.local_rank = args.local_rank if hasattr(args, 'local_rank') else 0
+        self.local_rank = args.local_rank if hasattr(
+            args,
+            'local_rank') else int(os.environ.get("LOCAL_RANK",
+                                                  -1))
+        env_local_rank = int(os.environ.get("LOCAL_RANK", -1))
+        if env_local_rank >= 0:
```

Comment:
I think it would be nice if this validation logic was moved into _do_args_sanity_check().